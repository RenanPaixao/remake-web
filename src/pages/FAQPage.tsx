import { Center, Heading } from '@chakra-ui/react'
import { QuestionsList } from '../components/QuestionsList/QuestionsList.tsx'
import { useTranslation } from 'react-i18next'

export const FAQPage = () => {
  const { t } = useTranslation()
  const contentList = [
    {
      title: t('faq.questions.how-to-separate'),
      content: t('faq.answers.how-to-separate')
    },
    {
      title: t('faq.questions.i-have-became-a-collector'),
      content: t('faq.answers.i-have-became-a-collector')
    },
    {
      title: t('faq.questions.how-many-places'),
      content: t('faq.answers.how-many-places')
    },
    {
      title: t('faq.questions.where-is-the-comment-section'),
      content: t('faq.answers.where-is-the-comment-section')
    }
  ]
  return <Center gap={8} pt={10} flexDirection={'column'}>
    <Heading as={'h1'}>FAQ</Heading>
    <QuestionsList minCardWidth={'22rem'} contentList={contentList} cardMargin={'2rem 0 0 0'}/>
  </Center>
}
