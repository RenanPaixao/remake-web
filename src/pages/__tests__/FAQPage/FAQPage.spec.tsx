import { expect } from 'vitest'
import { customRender } from '../../../../tests/test-utils.tsx'
import { FAQPage } from '../../FAQPage.tsx'

describe('FAQPage', () => {
  it('should render', async () => {
    const { container } = customRender(<FAQPage/>)

    expect(container).toMatchSnapshot()
  })
})
