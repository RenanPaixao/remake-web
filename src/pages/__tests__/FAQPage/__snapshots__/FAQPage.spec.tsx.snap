// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FAQPage > should render 1`] = `
<div>
  <div
    class="css-1bbbbuj"
  >
    <h1
      class="chakra-heading css-1dklj6k"
    >
      FAQ
    </h1>
    <div
      class="chakra-accordion css-0"
    >
      <div
        class="chakra-card chakra-accordion__item css-1kn1xtl"
      >
        <h2
          class="chakra-heading css-1dklj6k"
        >
          <button
            aria-controls="accordion-panel-:r0:"
            aria-expanded="false"
            class="chakra-accordion__button css-ynca4h"
            data-index="0"
            id="accordion-button-:r0:"
            type="button"
          >
            <span
              class="chakra-text css-1eziwv"
            >
              How to separate waste for recycling?
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r0:"
            class="chakra-accordion__panel css-z9zv9f"
            id="accordion-panel-:r0:"
            role="region"
          >
            Place separate bins for different types of waste in a convenient location in your home and Make sure everyone in your household understands the importance of recycling.
          </div>
        </div>
      </div>
      <div
        class="chakra-card chakra-accordion__item css-1kn1xtl"
      >
        <h2
          class="chakra-heading css-1dklj6k"
        >
          <button
            aria-controls="accordion-panel-:r2:"
            aria-expanded="false"
            class="chakra-accordion__button css-ynca4h"
            data-index="1"
            id="accordion-button-:r2:"
            type="button"
          >
            <span
              class="chakra-text css-1eziwv"
            >
              I've became a collector. How to add a place?
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r2:"
            class="chakra-accordion__panel css-z9zv9f"
            id="accordion-panel-:r2:"
            role="region"
          >
            Go to account page and click on "Become a collector" button. Then you will be able to add a place.
          </div>
        </div>
      </div>
      <div
        class="chakra-card chakra-accordion__item css-1kn1xtl"
      >
        <h2
          class="chakra-heading css-1dklj6k"
        >
          <button
            aria-controls="accordion-panel-:r4:"
            aria-expanded="false"
            class="chakra-accordion__button css-ynca4h"
            data-index="2"
            id="accordion-button-:r4:"
            type="button"
          >
            <span
              class="chakra-text css-1eziwv"
            >
              How many places can I add?
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r4:"
            class="chakra-accordion__panel css-z9zv9f"
            id="accordion-panel-:r4:"
            role="region"
          >
            You can add as many places as you want. We encourage you to add as many places as you can!
          </div>
        </div>
      </div>
      <div
        class="chakra-card chakra-accordion__item css-1kn1xtl"
      >
        <h2
          class="chakra-heading css-1dklj6k"
        >
          <button
            aria-controls="accordion-panel-:r6:"
            aria-expanded="false"
            class="chakra-accordion__button css-ynca4h"
            data-index="3"
            id="accordion-button-:r6:"
            type="button"
          >
            <span
              class="chakra-text css-1eziwv"
            >
              Where is the comments section about places present in the mobile application?
            </span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-:r6:"
            class="chakra-accordion__panel css-z9zv9f"
            id="accordion-panel-:r6:"
            role="region"
          >
            The comments about places is not present in the desktop application.
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
